clouddrift/adapters/gdp6h.py

```diff
@@ -77,15 +78,15 @@ def download(
 
     # retrieve all drifter ID numbers
     if drifter_ids is None:
-        urlpath = urllib.request.urlopen(url)
-        string = urlpath.read().decode("utf-8")
         drifter_urls: list[str] = []
         for dir in directory_list:
             urlpath = urllib.request.urlopen(os.path.join(url, dir))
             string = urlpath.read().decode("utf-8")
             filelist = list(set(re.compile(pattern).findall(string)))
             for f in filelist:
                 drifter_urls.append(os.path.join(url, dir, f))
+    else:
+        drifter_urls = [f"{url}/{filename_pattern.format(id=did)}" for did in drifter_ids]
```

Comment:
I don't think this can work, you are missing the `/dir/` here.

Reply:
I was actually going to push a modification that I just worked out, but since you change other stuff, I will just add it here. I replace this whole block with:

```
  # create list of drifter urls
  urlpath = urllib.request.urlopen(url)
  string = urlpath.read().decode("utf-8")
  drifter_urls: list[str] = []
  for dir in directory_list:
      urlpath = urllib.request.urlopen(os.path.join(url, dir))
      string = urlpath.read().decode("utf-8")
      filelist = list(set(re.compile(pattern).findall(string)))
      for f in filelist:
          if drifter_ids is None or int(f[:-3].split("_")[2]) in drifter_ids:
              drifter_urls.append(os.path.join(url, dir, f))
```